src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -265,9 +264,12 @@ private static string GetTypeAndHashSuffix(string exePath, bool isSingleFile)
                 {
                     try
                     {
-                        // Certain platforms may not have support for crypto
-                        hash = IdentityHelper.GetNormalizedUriHash(codeBase);
-                        typeName = UrlDesc;
+                        if (Uri.TryCreate(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, assembly.ManifestModule.Name), UriKind.Absolute, out Uri codeBase))
+                        {
+                            // Certain platforms may not have support for crypto
+                            hash = IdentityHelper.GetNormalizedUriHash(codeBase);
+                            typeName = UrlDesc;
+                        }
                     }
                     catch (PlatformNotSupportedException) { }
```

Comment:
Should the try/catch be inside the if (Uri...) block rather than outside of it?  It seems like the intention is it's targeted at the GetNormalizedUriHash call.

Reply:
Right